src/states/common/base.rs

```diff
@@ -70,6 +64,8 @@ pub trait Base: Display {
     fn unhandled_message(&mut self, sender: Option<SocketAddr>, msg: Message, msg_bytes: Bytes);
 
     fn handle_action(&mut self, action: Action, outbox: &mut dyn EventBox) -> Transition {
+        let _log_ident = self.set_log_ident();
```

Comment:
is this ident only works within the scope of `_log_ident` ?
which actually means for each incoming action/network_event, this `_log_ident` is a new instance?
and, what for `handle_message` ?

Reply:
Yes, it's a new instance every time, but it's a zero-sized type so there is no overhead. The actual buffer which stores the log ident (which is kept in a thread-local variable) is reused, so there is no allocation after the first couple of times.

> and, what for handle_message ?

`handle_message` is called in `handle_network_event` which also sets the log ident.